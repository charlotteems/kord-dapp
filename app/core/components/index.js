import AccountForm from './AccountForm'
import Link from './Link'
import Logo from './Logo'

export { AccountForm, Link, Logo }
