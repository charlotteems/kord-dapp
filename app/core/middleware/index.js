import { createLogger } from 'redux-logger'
import { middleware as reduxPack } from 'redux-pack'

import SessionMiddleware from './session'
import UIMiddleware from './ui'

const createMiddleware = isDevelopment => {
  // default middleware
  const middleware = [reduxPack, SessionMiddleware, UIMiddleware]

  // logger middleware in development
  if (isDevelopment) middleware.push(createLogger({ collapsed: true }))

  return middleware
}

export default createMiddleware
