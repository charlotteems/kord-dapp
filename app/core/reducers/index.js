import { foundReducer } from 'found'
import { combineReducers } from 'redux-immutablejs'

import * as Claims from 'domains/claims'
import * as Identity from 'domains/identity'
import * as MarketPlace from 'domains/marketplace'
import * as Profile from 'domains/profile'
import * as Session from 'domains/session'
import * as UI from 'domains/ui'

export default combineReducers({
  found: foundReducer,
  [Claims.name]: Claims.reducer,
  [Identity.name]: Identity.reducer,
  [MarketPlace.name]: MarketPlace.reducer,
  [Profile.name]: Profile.reducer,
  [Session.name]: Session.reducer,
  [UI.name]: UI.reducer,
})
