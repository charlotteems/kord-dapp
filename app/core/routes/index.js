import React from 'react'
import { Route, makeRouteConfig } from 'found'

import App from 'core/containers/App'
import routes from 'core/routes/config'
import * as pages from 'pages'

export { routes }

export default makeRouteConfig(
  <Route Component={App}>
    <Route Component={pages.Home} path={routes.home.path} />
  </Route>
)
