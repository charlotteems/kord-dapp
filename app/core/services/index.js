import * as Facebook from './facebook'
import * as Fetch from './fetch'
import * as KordId from './kord-id'
import * as KordNetwork from './kord-network'
import Sentry from './sentry'
import * as Swarm from './swarm'
import * as WebStorage from './web-storage'

export { Facebook, Fetch, KordId, KordNetwork, Sentry, Swarm, WebStorage }
