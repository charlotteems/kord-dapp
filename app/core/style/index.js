import * as animation from './animation'
import customTheme from './theme'

export { animation, customTheme }
