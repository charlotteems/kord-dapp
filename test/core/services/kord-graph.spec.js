import { generate } from 'ethereumjs-wallet'
import { bufferToHex } from 'ethereumjs-util'

import { KordGraph } from 'core/services'
import createClaimVariables from './fixtures/create-claim-variables.json'
import createGraphVariables from './fixtures/create-graph-variables.json'
import readClaimsByGraphVariables from './fixtures/read-claims-graph-variables.json'
import setGraphVariables from './fixtures/set-graph-variables.json'

jest.mock('core/services/kord-network')

describe('core/services/kord-graph/createGraph', () => {
  it('Should return successful mutation response', async () => {
    // generate a new public address
    const address = bufferToHex(generate().getAddress())

    // update test data to avoid duplicate graph error
    createGraphVariables.graph.id = address

    const actual = await KordGraph.createGraph(createGraphVariables)

    // anonymize API response to match snapshot
    actual.data.createGraph.id = '0x0000000000000000000000000000000000000000'
    actual.extensions.meta.swarmHash =
      '0x0000000000000000000000000000000000000000000000000000000000000000'

    expect(actual).toMatchSnapshot()
  })
})

describe('core/services/kord-graph/setGraph', () => {
  it('Should return successful mutation response', async () => {
    const actual = await KordGraph.setGraph(setGraphVariables)

    // anonymize API response to match snapshot
    actual.setGraph.id = '0x0000000000000000000000000000000000000000'

    expect(actual).toMatchSnapshot()
  })
})

describe('core/services/kord-graph/createClaim', () => {
  it('Should return successful mutation response', async () => {
    const actual = await KordGraph.createClaim(createClaimVariables)

    // anonymize API response to match snapshot
    actual.extensions.meta.swarmHash =
      '0x0000000000000000000000000000000000000000000000000000000000000000'

    expect(actual).toMatchSnapshot()
  })
})

describe('core/services/kord-graph/readClaimsByGraph', () => {
  it('Should return query response', async () => {
    const actual = await KordGraph.readClaimsByGraph(readClaimsByGraphVariables)

    expect(actual).toHaveProperty('graph.id')
    expect(actual).toHaveProperty('graph.claim')
  })
})
