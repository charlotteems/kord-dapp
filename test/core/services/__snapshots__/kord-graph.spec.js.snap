// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/services/kord-graph/createClaim Should return successful mutation response 1`] = `
Object {
  "createClaim": Object {
    "claim": "test1",
    "id": "0x420d65042304f05a86964bb2ba68bb863c34ae5363228370254d83a47235d6d8",
    "issuer": "0xEe078019375fBFEef8f6278754d54Cf415e83329",
    "property": "test",
    "signature": "0xbe8ad51e928b5fcb461543e8efe9666f0a7309f3d0a54a644f089de48e09664e0ee6dc62fcd6e0aed66c5ba954b4066245f600bcc17e49b4e05c2973bdaf571c01",
    "subject": "0xEe078019375fBFEef8f6278754d54Cf415e83329",
  },
}
`;

exports[`core/services/kord-graph/setGraph Should return successful mutation response 1`] = `
Object {
  "setGraph": Object {
    "id": "0xEe078019375fBFEef8f6278754d54Cf415e83329",
  },
}
`;
